services/tracks/src/services/trackService.js

```diff
@@ -3,8 +3,22 @@
 const { Tracks } = require('../db/models/Tracks');
 const { v4: uuid } = require('uuid');
 const { axiosClient } = require('../config/axios.config');
+const { kafkaTopics } = require('../constants');
+
+class TrackService {
+  notificationProducer;
+
+  constructor(notificationProducer) {
+    this.notificationProducer = notificationProducer;
+    (async () => {
+      try {
+        await this.notificationProducer.connect();
+      } catch (e) {
```

Comment:
As I saw in your code, you mostly use `err` variable name for errors, so I guess you should change it here